File: app/src/main/java/com/personal/accountantAssistant/utils/DialogUtils.java
package com.personal.accountantAssistant.utils;

import android.app.AlertDialog;
import android.content.Context;

import com.personal.accountantAssistant.R;

import java.util.concurrent.atomic.AtomicInteger;

import io.reactivex.functions.Action;

public class DialogUtils {

    public static void confirmationDialog(final Context context,
                                          final int titleId,
                                          final int messageId,
                                          final Action confirmAction) {
        confirmationDialog(context,
                context.getString(titleId),
                context.getString(messageId),
                confirmAction,
                ActionUtils.NONE_ACTION_TO_DO);
    }

    private static void confirmationDialog(final Context context,
                                           final String title,
                                           final String message,
                                           final Action confirmAction,
                                           final Action cancelAction) {
        new AlertDialog.Builder(context)
                .setTitle(title)
                .setMessage(message)
                .setPositiveButton(R.string.ok, (dialog, id) -> ActionUtils.runAction(confirmAction))
                .setNegativeButton(R.string.cancel, (dialog, id) -> ActionUtils.runAction(cancelAction))
                .show();
    }

    public static void showImportExportDialog(final Context context,
                                              final int titleId,
                                              final Action importAction,
                                              final Action exportAction) {
        showImportExportDialog(context,
                context.getString(titleId),
                importAction,
                exportAction);
    }

    private static void showImportExportDialog(final Context context,
                                               final String title,
                                               final Action importAction,
                                               final Action exportAction) {
        final int IMPORT_OPTION = 0;
        final int EXPORT_OPTION = 1;
        final String[] options = {
                context.getString(R.string.import_all),
                context.getString(R.string.export_all)
        };
        final AtomicInteger checkedItem = new AtomicInteger(0);

        new AlertDialog.Builder(context)
                .setTitle(title)
                .setSingleChoiceItems(options, checkedItem.get(), (dialog, which) -> checkedItem.set(which))
                .setPositiveButton(R.string.ok, (dialog, id) -> {
                    switch (checkedItem.get()) {
                        case IMPORT_OPTION:
                            ActionUtils.runAction(importAction);
                            break;
                        case EXPORT_OPTION:
                            ActionUtils.runAction(exportAction);
                            break;
                    }
                })
                .setNegativeButton(R.string.cancel, (dialog, id) -> {
                })
                .show();
    }
}

File: app/src/main/java/com/personal/accountantAssistant/utils/DialogUtils.kt
package com.personal.accountantAssistant.utils

import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface
import com.personal.accountantAssistant.R
import com.personal.accountantAssistant.utils.ActionUtils.runAction
import io.reactivex.functions.Action
import java.util.concurrent.atomic.AtomicInteger

object DialogUtils {

    fun confirmationDialog(
        context: Context,
        titleId: Int,
        messageId: Int,
        confirmAction: Action
    ) {
        confirmationDialog(
            context,
            context.getString(titleId),
            context.getString(messageId),
            confirmAction,
            ActionUtils.NONE_ACTION_TO_DO
        )
    }

    private fun confirmationDialog(
        context: Context,
        title: String,
        message: String,
        confirmAction: Action,
        cancelAction: Action
    ) {
        AlertDialog.Builder(context)
            .setTitle(title)
            .setMessage(message)
            .setPositiveButton(R.string.ok) { _: DialogInterface?, _: Int -> runAction(confirmAction) }
            .setNegativeButton(R.string.cancel) { _: DialogInterface?, _: Int ->
                runAction(
                    cancelAction
                )
            }
            .show()
    }

    fun showImportExportDialog(
        context: Context?,
        titleId: Int,
        importAction: Action,
        exportAction: Action
    ) {
        context?.let {
            showImportExportDialog(
                it,
                it.getString(titleId),
                importAction,
                exportAction
            )
        }
    }

    private fun showImportExportDialog(
        context: Context,
        title: String,
        importAction: Action,
        exportAction: Action
    ) {
        val importOption = 0
        val exportOption = 1
        val options = arrayOf(
            context.getString(R.string.import_all),
            context.getString(R.string.export_all)
        )
        val checkedItem = AtomicInteger(0)
        AlertDialog.Builder(context)
            .setTitle(title)
            .setSingleChoiceItems(
                options,
                checkedItem.get()
            ) { _: DialogInterface?, which: Int -> checkedItem.set(which) }
            .setPositiveButton(R.string.ok) { _: DialogInterface?, _: Int ->
                when (checkedItem.get()) {
                    importOption -> runAction(importAction)
                    exportOption -> runAction(exportAction)
                }
            }
            .setNegativeButton(R.string.cancel) { _: DialogInterface?, _: Int -> }
            .show()
    }
}